File: src/main/java/com/nextspring/modules/category/CategoryController.java
package com.nextspring.modules.category;

import java.util.UUID;

import org.springframework.graphql.data.method.annotation.Argument;
import org.springframework.graphql.data.method.annotation.MutationMapping;
import org.springframework.graphql.data.method.annotation.QueryMapping;
import org.springframework.stereotype.Controller;

import lombok.RequiredArgsConstructor;

@Controller
@RequiredArgsConstructor
public class CategoryController {

	private final CategoryService categoryService;

	@MutationMapping
	CategoryEntity addCategory(@Argument CategoryInput category) {
		return categoryService.addCategory(category);
	}

	@QueryMapping
	Iterable<CategoryEntity> getCategories() {
		return categoryService.getCategories();
	}

	@QueryMapping
	CategoryEntity findCategoryById(@Argument UUID id) {
		return categoryService.findCategoryById(id);
	}

	@MutationMapping
	CategoryEntity updateCategory(@Argument UUID id, @Argument CategoryInput category) {
		return categoryService.updateCategory(id, category);
	}

	@MutationMapping
	void deleteCategory(@Argument UUID id) {
		categoryService.deleteCategory(id);
	}

}

File: src/main/kotlin/com/nextspring/modules/category/CategoryController.kt
package com.nextspring.modules.category

import lombok.RequiredArgsConstructor
import org.springframework.graphql.data.method.annotation.Argument
import org.springframework.graphql.data.method.annotation.MutationMapping
import org.springframework.graphql.data.method.annotation.QueryMapping
import org.springframework.stereotype.Controller
import java.util.*

@Controller
@RequiredArgsConstructor
class CategoryController(
    private val categoryRepository: CategoryRepository
) {

    @MutationMapping
    fun addCategory(@Argument input: CategoryInput): CategoryEntity {
        val categoryEntity = CategoryEntity(UUID.randomUUID(), input.name)

        return categoryRepository.save(categoryEntity)
    }

    @QueryMapping
    fun categories(): Iterable<CategoryEntity?> {
        return categoryRepository.findAll()
    }

    @QueryMapping
    fun category(@Argument id: UUID): CategoryEntity {
        return categoryRepository.findById(id).orElseThrow()!!
    }

    @QueryMapping
    fun updateCategory(@Argument id: UUID): CategoryEntity {
        return categoryRepository.findById(id).orElseThrow()!!
    }

    @MutationMapping
    fun deleteCategory(@Argument id: UUID) {
        categoryRepository.deleteById(id)
    }

}
